src/rt/cast_.d

```diff
@@ -14,6 +14,9 @@
 module rt.cast_;
 
 extern (C):
+@nogc:
+nothrow:
+pure:
```

Comment:
Adding some attributes here so the compiler can verify the attributes I claim in the `private extern(C)` declaration in `object.d`.